reference/src/functions.md

```diff
@@ -147,6 +147,33 @@ module a::m_test {
 }
 ```
 
+### `macro` modifier
+
+Unlike normal functions, `macro` functions do not exist at runtime. Instead, these functions are
+substituted inline at each call site during compilation. These `macro` functions leverage this
+compilation process to provide functionality beyond standard functions, such as accepting
+higher-order _lambda_-style functions as arguments. These lambda arguments, also expanded
+during compilation, allow you to pass parts of the function body to the macro as  arguments.
+For instance, consider the following simple loop macro, where the loop body is supplied as a lambda:
+
+```move
+macro fun ntimes($n: u64, $body: |u64| -> ()) {
+    let n = $n;
+    let mut i = 0;
+    while (i < n) {
+        $body(i);
+        i = i + 1;
+    }
+}
+
+fun example() {
+    let mut sum = 0;
+    ntimes!(10, |i| sum = sum + i );
```

Comment:
Do we want to use a different variable name in lambda definition to avoid implying that it has to be somehow the same as what's used in the macro body?